runtime/caml/domain.h

```diff
@@ -121,6 +121,7 @@ int caml_global_barrier_is_final(barrier_status);
 void caml_global_barrier_end(barrier_status);
 int caml_global_barrier_num_domains(void);
 
+int caml_domain_terminating(caml_domain_state *);
 int caml_domain_is_terminating(void);
```

Comment:
I wonder whether we can do better with the function names. Given that we now have `caml_domain_terminating` it is not obvious that `caml_domain_is_terminating` refers to the current domain. 

For example, the only place where `caml_domain_is_terminating` is at

https://github.com/ocaml/ocaml/blob/fd0551b7b73e9b65e105a7071642aae155d1ae77/runtime/major_gc.c#L399-L406

It may be best to remove `caml_domain_is_terminating` and only use `caml_domain_terminating` (which would be better named as `caml_domain_is_terminating`.

Reply:
I would be happy with `caml_domain_is_terminating` having an argument, but there are several `caml_domain_is_foo` functions and they all implicitly talk about the current/self domain. (`caml_domain_is_in_multicore`, `caml_domain_is_in_stw`; and also... `caml_domaine_alone ). I didn't want to break the regularity there, so I did the so-so choice of having a minor variant of the function that takes an argument.

I am happy to discuss better naming schemes for these three functions together, but I think this can be done independently of whether the present PR is merged.

For example we could rename those functions into `caml_current_domain_is_foo`, and `caml_domain_is_foo` would take a parameter. Or we could just give up on the `void` version and always pass the domain explicitly. (But I'm not sure they all take a `domain_state`, maybe some rely on `domain_self` instead.)

Reply:
(Passing the domain explicitly would also sometimes be slightly more efficient as it avoids loading the thread-local variable again.)

Reply:
I proposed something on that space in #12617.